lldb/source/Plugins/Language/CPlusPlus/CxxStringTypes.cpp

```
@@ -68,11 +68,9 @@ static bool CharStringSummaryProvider(ValueObject &valobj, Stream &stream) {
 
 template <StringElementType ElemType>
 static bool CharSummaryProvider(ValueObject &valobj, Stream &stream) {
-  DataExtractor data;
-  Status error;
-  valobj.GetData(data, error);
+  auto data_or_err = valobj.GetData();
 
-  if (error.Fail())
+  if (!data_or_err)
```

The important difference between llvm::Error is that it asserts if you don't check the error inside. So you need to explicitly throw away the error here.

In situations where the error really doesn't matter, you can wrap it in `llvm::expectedToOptional()` which will discard the error and return a `std::optional`. In other cases, logging the error is appropriate, for example: `LLDB_LOG_ERRORV(GetLog(LLDBLog::Formatters), data_or_err.takerError(), "Cannot extract data: {0}")`.
And then some functions returns an llvm::Error or a Status, in which case you can just pass the error up.

I think this function would be a candidate for logging.

---

Okay that makes sense, I was aware of the assertion of `llvm::Error`. I'll make sure to fix that and resort to handling the Errors if needed so. Otherwise, I'll use `llvm::expectedToOptional()`. Just to be sure, in what case would I know if a error doesn't really matter?